### ISSUE TITLE
Ability to skip filesystems basing on sender's configuration

### ISSUE THREAD

**Issue by mateuszkwiatkowski** [zrepl/zrepl#191]
It would be great to be able to skip filesystems from backup basing on sender's configuration. For example on ZFS attributes (example: `zfs set zrepl:skip=true zroot/vm/myvirtualmachine`).
The use case is that I want to be able to disable/enable this attribute from virtual machines control panel.

**Comment by problame** [comment]
I agree with the idea that the pull source should be able to pre-filter what it presents to the pull jobs  (I guess it's a pull setup in your scenario).
In fact, this feature has been present once, but I discarded it because it complicated the sync logic at the time.

However, I'm not sure how I feel about zrepl configuration in ZFS properties.
The main problem I have with them is that in theory, multiple jobs could be pulling from one source.
Thus, you'd need `zrepl:JOBNAME:configurationvalue` as a property.
And then, users start renaming the job, which brings config.yml and ZFS properties out of sync.
Plus, policy is now spread over lots of dataset attributes as opposed to one file.

Would you like to tell me what VM control panel you are using? Proxmox?

**Comment by mateuszkwiatkowski** [comment]
Hi Problame,
Many thanks for your response.
You’re right that we’re using pull setup.
I think that in our simple use case, ability to omit a filesystem from all jobs would be fine - that means that single ZFS property would be fine.
I think we can also have a conf.d style configuration with files that list file systems to skip. Maybe this is already supported?
I use and develop Runhyve - manager for bhyve on FreeBSD.

**Comment by problame** [comment]
Wow, pretty impressed by runhyve!
>  I think we can also have a conf.d style configuration with files that list file systems to skip. Maybe this is already supported?

Not sure what purpose a conf.d file would serve compared to rendering a new config and restarting zrepl.
conf.d is not supported ATM.

Do you want to integrate zrepl into the hypervisor or just provide users generic means to set dataset properties?

**Comment by mateuszkwiatkowski** [comment]
Thank you. It’s still in its infa
So for conf.d approach dynamic config reload is needed. In that case use of ZFS properties will be more convenient.
